chronos/internal/asyncengine.nim

```diff
@@ -10,6 +10,92 @@
 
 {.push raises: [].}
 
+## This module implements the core asynchronos engine / dispatcher.
+##
+## The dispatcher acts as a kind of event loop. You must call ``poll`` on it
+## (or a function which does so for you such as ``waitFor`` or ``runForever``)
+## in order to poll for any outstanding events. The underlying implementation
+## is based on epoll on Linux, IO Completion Ports on Windows and select on
+## other operating systems.
+##
+## The ``poll`` function will not, on its own, return any events. Instead
+## an appropriate ``Future`` object will be completed. A ``Future`` is a
```

Comment:
It might make sense to include *how* a poll call connects to a Future completing.
Or link to a source that provides a mental image that makes that connection.

At least this tripped me up from my very beginner's perspective on the matter ([discord link](https://discord.com/channels/371759389889003530/371759389889003532/1172642686994493532) to my being confused).

Reply:
removed - concept docs explain this better